src/Microsoft.Azure.WebJobs.ServiceBus/EventHubs/EventHubConfiguration.cs

```diff
@@ -445,12 +445,12 @@ void IExtensionConfigProvider.Initialize(ExtensionConfigContext context)
                 .AddConverter<EventData, byte[]>(ConvertEventData2Bytes);
 
             // register the background exception handler
-            MessagingExceptionHandler.Subscribe(_options, context.Trace, context.Config.LoggerFactory);
+            var exceptionHandler = MessagingExceptionHandler.Subscribe(_options, context.Trace, context.Config.LoggerFactory);
```

Comment:
This was a memory leak as well? I had considered this issue when I authored the code originally - see my comments in https://github.com/Azure/azure-webjobs-sdk/pull/1788/files.

Reply:
It certainly looked like it was -- I saw ScriptHosts hanging around and saw this in their GC roots. Let me take another look, though. Thanks for calling it out.

Reply:
I removed that change and ran through my scenario again to double-check -- this time with just one function with an EH Trigger. 

The ScriptHosts and EventHubListeners stay around and the roots in VS look like below. When I unsubscribe, things get cleaned up like I'd expect.

![image](https://user-images.githubusercontent.com/1089915/47223501-4c9cae00-d36e-11e8-8bc3-77c6e3cb5687.png)